lib/agent.lua

```diff
@@ -94,15 +102,35 @@ function Agent:start(options)
       if not options.pidFile then
         options.pidFile = constants:get('DEFAULT_PID_FILE_PATH')
       end
-      --err, msg = virgo.write_pid(options.pidFile)
-      --if err then
-      --  pcall(function()
-      --    local pid = fs.readFileSync(options.pidFile)
-      --    logging.error(fmt('Agent in use (pid: %d, path: %s)', pid, options.pidFile))
-      --    process:exit(3)
-      --  end)
-      --end
-      callback()
+      fs.open(options.pidFile, 'a+', function(err, fd)
```

Comment:
'w' does truncate the log file... regardless of if you have the lock or not.  that's why many programs use both a lock file (contents are usually empty) and a separate pid file.